File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    let answer = 0;
    
    const memo = {};
    let count = 0;
    
    nums.forEach(num => {
        if (memo[num] === undefined) {
            memo[num] = true;
            ++count;
        }
    });
    
    if (count > nums.length / 2) {
        answer = nums.length / 2;
    } else {
        answer = count;
    }
    
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <set>
#include <algorithm>

#include <iostream>

using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    
    int numCount = 0;
    set<int> record;
    
    for (int num : nums)
    {
        if (record.find(num) == record.end())
        {
            record.insert(num);
            ++numCount;
        }
    }
    
    answer = min((int)(nums.size() / 2), numCount);
    
    
    return answer;
}